main.cpp

```diff
@@ -18,13 +18,27 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
-public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    Rewriter& rewrite;
+ public:
+    CastCallBack(Rewriter& rewriter) rewrite(rewriter) {};
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        const auto *expr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+        if (expr != nullptr) {
+            auto loc = CharSourceRange::getCharRange(
+                expr->getLParenLoc(), expr->getSubExprAsWritten()->getBeginLoc());

```

Comment:
Please describe the difference between `getSubExprAsWritten()->getBeginLoc()` and `getRParenLoc().getLocWithOffset(1)` with examples.

Reply:
The matter is, that unlike `getRParenLoc().getLocWithOffset(1)`, `getSubExprAsWritten()->getBeginLoc()` gets us the location of the first symbol in the name of the variable we need to cast, thus ignoring any characters between it and the closing parenthesis after the type name we need to cast this variable to, while `getRParenLoc().getLocWithOffset(1)` straight up gets us the first character after the before-mentioned type name.

With this in mind, these functions work as follows:
```
//This is a code sample
int main() {
    int var;
    double i = (double)                     var;
    return 0;
}
```

```
//This is an example of getRParenLoc().getLocWithOffset(1)
int main() {
    int var;
    double i = static_cast<double>(                     var);
    return 0;
}
```

```
//This is an example of getSubExprAsWritten()->getBeginLoc()
int main() {
    int var;
    double i = static_cast<double>(var);
    return 0;
}
```